modules/contact/src/actions/ExplicitDynamicsContactAction.C

```diff
@@ -0,0 +1,442 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ExplicitDynamicsContactAction.h"
+
+#include "Factory.h"
+#include "FEProblem.h"
+#include "Conversion.h"
+#include "AddVariableAction.h"
+#include "NonlinearSystemBase.h"
+#include "Parser.h"
+
+#include "NanoflannMeshAdaptor.h"
+#include "PointListAdaptor.h"
+
+#include <set>
+#include <algorithm>
+#include <unordered_map>
+#include <limits>
+
+#include "libmesh/petsc_nonlinear_solver.h"
+#include "libmesh/string_to_enum.h"
+
+// Counter for naming auxiliary kernels
+static unsigned int ed_contact_auxkernel_counter = 0;
```

Comment:
I think the preferred pattern is to build the object names to include the action name to disambiguate. This leads to more informative object names and obviates the need for statics here. But I can make that change later on.